### PR TITLE
added asking for rain status

### PR DESCRIPTION
**Pull request by FlxB2**
small weather extension - its now possible to ask things like
"is it gonna rain tomorrow in berlin"

Pull request status:
- [ ] Tests added/updated
- [ ] 80 % Coverage
- [x] Code conventions
- [x] Doku required and added
- [x] Ready for review

### CODE REVIEW

File: plugins/weather/src/main/resources/WeatherSpeech.aim.xml [hunk 1]
```diff
@@ -16,6 +16,18 @@
 		ref="de.unistuttgart.iaas.amyassist.amy.plugin.weather.WeatherSpeechCommand.weatherWeekend">
 		<gram>[how is the] weather [at the] weekend</gram>
 	</Intent>
+	<Intent
+		ref="de.unistuttgart.iaas.amyassist.amy.plugin.weather.WeatherSpeechCommand.rainCheck">
+		<gram>is it gonna rain {timespan} [(at|in) {locationname}]</gram>
+		<EntityTemplates>
+			<EntityTemplate id="timespan" required="false">
+				<gram>(today|tomorrow|at the weekend)</gram>
```
Comment (sinpat) [inline plugins/weather/src/main/resources/WeatherSpeech.aim.xml]: `On the weekend` would be the proper term

Comment (Bensge) [reply plugins/weather/src/main/resources/WeatherSpeech.aim.xml]: I think `{timespan}` should be at the end of the sentence. Is that possible? Not sure if it conflicts with the wildcard or +

Comment (sinpat) [reply plugins/weather/src/main/resources/WeatherSpeech.aim.xml]: Yeah you are right, putting it at the end makes the sentence more natural

**Comment by sinpat** [comment]
When there is a 0% chance of precipitation, the output is: `0% chance of null today`

**Comment by FlxB2** [comment]
> When there is a 0% chance of precipitation, the output is: `0% chance of null today`

Ah forgot about that, will fix it. Thanks!

File: plugins/weather/src/main/resources/WeatherSpeech.aim.xml (hunk 1 above)
Comment (FlxB2) [reply plugins/weather/src/main/resources/WeatherSpeech.aim.xml]: I agree, i'll try to change it

**Comment by FlxB2** [comment]
waiting for #482

**Comment by Bensge** [comment]
@FlxB2 Is this ready for review now?

**Comment by FlxB2** [comment]
@Bensge i'll finish and test this tonight.

**Comment by FlxB2** [comment]
fixed with merge

File: plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherSpeechCommand.java [hunk 2]
```diff
@@ -266,6 +267,17 @@ public String rainCheck(Map<String, EntityData> entities) {
 
 		return "i don't understand the date";
 	}
+	
+	private String stringifyRainCheck(double probability, String precipType, String suffix) {
+		final String chanceOf = "% chance of ";
+
+		if(round(probability) == 0) {
+			return "0% chance of rain " + suffix + "!";
```
Comment (Bensge) [inline plugins/weather/src/main/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherSpeechCommand.java]: I think at 0% probability, the DS API returns "no-data" as the precip type. Maybe respond with "No precipitation today/tomorrow" instead

**Pull request merged**